vllm/worker/model_runner.py

```diff
@@ -916,27 +963,80 @@ def capture_model(self, kv_caches: List[torch.Tensor]) -> None:
             bs for bs in _BATCH_SIZES_TO_CAPTURE if bs <= graph_batch_size
         ]
 
+        # For flashinfer, different batch sizes will share the same workspace buffer
+        decode_workspace_buffer = torch.empty(128 * 1024 * 1024,
+                                                dtype=torch.uint8,
+                                                device=self.device)
         with graph_capture() as graph_capture_context:
             # NOTE: Capturing the largest batch size first may help reduce the
             # memory usage of CUDA graph.
             for batch_size in reversed(batch_size_capture_list):
+                indptr_buffer = torch.empty(batch_size + 1,
+                                            dtype=torch.int32,
```

Comment:
QQ: how is dtype decided? The doc doesn't seem to specify the dtype https://docs.flashinfer.ai/api/python/decode.html

Reply:
I got it from flashinfer tests 😂
https://github.com/flashinfer-ai/flashinfer/blob/main/python/tests/test_batch_decode_kernels.py#L164

Reply:
AFAIK this dtype doesn't really matter because flashinfer kernel would just need a 32-bit dtype buffer in its kernel.